server.js

```diff
@@ -42,8 +42,22 @@ const searchResults = [
 // 1. Starts with query.
 // 2. Any word starts with query.
 // 3. Any word starts with all space-separated query.
-app.post('/search', function(req, res) {
-  return res.json([]);
-});
+app.post('/search', function (req, res) {
+  return res.json(searchResults.filter(word => {
+    let wordArr = word.split(' ');
+    let searchTerms = req.body.searchTerm.toLowerCase().split(' ');
+
+    let res = true;
+    for (let elem of searchTerms) {
+      res = res && wordArr.some(w => {
```

Comment:
You can simplify this by checking if `wordArr.some()` returns false and if it does, you can `return false;` and immediately exit the for loop.